core/serialization/deserializer/montage-interpreter.js

```
@@ -170,8 +172,10 @@ var MontageContext = Montage.specialize({
             }
 
             if (promises.length === 0) {
-                return Promise.resolve(this._invokeDidReviveObjects());
+                result = this._invokeDidReviveObjects();
```

How is that working if the for() above happens to fill promises array while the interpreter has been set to be synchronous?

---

Because `getObject` will throw an error instead of returning a promise if the context has been set to sync but some async work needs to be done.

---

Because `getObject` will throw an error instead of returning a promise if the context has been set to sync but some async work needs to be done.